src/Application/Actions/Hooks/StripeUpdate.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MatchBot\Application\Actions\Hooks;
+
+use Doctrine\ORM\EntityManagerInterface;
+use MatchBot\Application\Actions\Action;
+use MatchBot\Application\Actions\ActionError;
+use MatchBot\Application\Actions\ActionPayload;
+use MatchBot\Domain\Donation;
+use MatchBot\Domain\DonationRepository;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Serializer\SerializerInterface;
+use Stripe\Event as StripeEvent;
+
+/**
+ * @return Response
+ */
+class StripeUpdate extends Action
+{
+    private DonationRepository $donationRepository;
+    private EntityManagerInterface $entityManager;
+    private SerializerInterface $serializer;
+
+    public function __construct(
+        DonationRepository $donationRepository,
+        EntityManagerInterface $entityManager,
+        LoggerInterface $logger,
+        SerializerInterface $serializer
+    ) {
+        $this->donationRepository = $donationRepository;
+        $this->entityManager = $entityManager;
+        $this->serializer = $serializer;
+
+        parent::__construct($logger);
+    }
+
+    protected function action(): Response
+    {
+        $payload = $this->request->getBody();
+        $signature = $this->request->getHeaderLine('stripe-signature');
+        $webhookSecret = getenv('STRIPE_WEBHOOK_SIGNING_SECRET');
```

Comment:
@tbg-jasonfung as we already have a `'stripe'` key in the config object I think we should put this there too so it's obvious where to look for configs, but can do this in a later update.